### PR TITLE
Ensure -WhatIf parameter is honored by all scripts commands and fix hybrid disconnected alert bug

### PR DESCRIPTION
**Pull request by Brunoga-MS**
# Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. *Fixes a bug in the Deploy-Hybrid-VM-Disconnected-Alert.json policy definition that was not using the autoMitigate template value [workitem #37038]
(https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_sprints/backlog/Azure%20Landing%20Zones%20AMBA%20Team/Azure%20Landing%20Zones/Dilithium/202409?workitem=37038)*

image

2. *Fixes a bug in the Start-AMBACleanup.ps1 that prevented the UAMI to be correctly removed [workitem #37039(https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_sprints/backlog/Azure%20Landing%20Zones%20AMBA%20Team/Azure%20Landing%20Zones/Dilithium/202409?workitem=37039)*
3. *Ensures the _WhatIf_ parameter is honored by all the cmdlets used in AMBA-ALZ scripts [work item #](https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_sprints/backlog/Azure%20Landing%20Zones%20AMBA%20Team/Azure%20Landing%20Zones/Dilithium/202409)*
4. Updates the related documentation 

### Breaking Changes

1. *NONE*

## As part of this Pull Request I have

- [X] Read the Contribution Guide and ensured this PR is compliant with the guide
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/azure-monitor-baseline-alerts/pulls)
- [X] Associated it with relevant [GitHub Issues](https://github.com/Azure/azure-monitor-baseline-alerts/issues) or ADO Work Items (Internal Only)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/azure-monitor-baseline-alerts/)
- [X] Ensured PR tests are passing
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2023-11-14.md [hunk 1]
```diff
@@ -25,20 +25,20 @@ Updating from release [2023-11-14](../../Whats-New#2023-11-14) will require runn
 
   {{% include "PowerShell-ExecutionPolicy.md" %}}
 
-  **Generate a list of the resource IDs which would be deleted by this script:**
+  **Show output of what would happen if deletes executed:**
 
   ```powershell
-  ./Start-AMBAOldArpCleanup.ps1 -pseudoRootManagementGroup $pseudoRootManagementGroup -ReportOnly
+  ./Start-AMBAOldArpCleanup.ps1 -pseudoManagementGroup $pseudoRootManagementGroup -WhatIf
```
Comment (arjenhuitema) [inline docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2023-11-14.md:31]: Can you validate? Looks like the param of the script is `[string]$pseudoRootManagementGroup`

File: docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2024-03-01.md [hunk 2]
```diff
@@ -28,20 +28,20 @@ To run the script, complete the following step:
 
   {{% include "PowerShell-ExecutionPolicy.md" %}}
 
-  **Generate a list of the resource IDs which would be deleted by this script:**
+  **Show output of what would happen if deletes executed:**
 
   ```powershell
-  ./Remove-AMBANotificationAssets.ps1 -pseudoRootManagementGroup $pseudoRootManagementGroup -ReportOnly
+  ./Remove-AMBANotificationAssets.ps1 -pseudoManagementGroup $pseudoRootManagementGroup -WhatIf
```
Comment (arjenhuitema) [inline docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2024-03-01.md:34]: Can you validate? Looks like the param of the script is `[string]$pseudoRootManagementGroup`

File: docs/content/patterns/alz/Cleaning-up-a-Deployment.md [hunk 3]
```diff
@@ -39,29 +39,29 @@ Follow the instructions below to download the cleanup script file. Alternatively
...
 6. Execute the script using one of the options below:
 
     {{% include "PowerShell-ExecutionPolicy.md" %}}
 
-    **Generate a list of the resource IDs which would be deleted by this script:**
+    **Show output of what would happen if deletes executed:**
 
     ```powershell
-    ./Start-AMBACleanup.ps1 -pseudoRootManagementGroup $pseudoRootManagementGroup -ReportOnly
+    ./Start-AMBACleanup.ps1 -pseudoManagementGroup $pseudoRootManagementGroup -WhatIf
```
Comment (arjenhuitema) [inline docs/content/patterns/alz/Cleaning-up-a-Deployment.md:54]: Can you validate? Looks like the param of the script is `[string]$pseudoRootManagementGroup`

File: patterns/alz/scripts/Remove-AMBADeployments.ps1 [hunk 4]
```diff
@@ -12,37 +33,29 @@
     https://github.com/Azure/azure-monitor-baseline-alerts
 
 .EXAMPLE
-    ./Remove-AMBADeployments.ps1 -pseudoManagementGroup Contoso -ReportOnly
-    # generate a list of the deployments which would be deleted by this script
+    ./Remove-AMBADeployments.ps1 -pseudoManagementGroup Contoso -WhatIf
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Remove-AMBADeployments.ps1:36]: Please validate param in examples.

File: patterns/alz/scripts/Remove-AMBANotificationAssets.ps1 [hunk 5]
```diff
@@ -14,36 +25,31 @@
...
 
     This script leverages the Azure Resource Graph to find object to delete. Note that the Resource Graph lags behind ARM by a couple minutes.
+
 .LINK
     https://github.com/Azure/azure-monitor-baseline-alerts
 
 .EXAMPLE
-    ./Remove-AMBANotificationAssets.ps1 -pseudoManagementGroup Contoso -ReportOnly
-    # generate a list of the resource IDs which would be deleted by this script
+    ./Remove-AMBANotificationAssets.ps1 -pseudoManagementGroup Contoso -WhatIf
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Remove-AMBANotificationAssets.ps1:33]: Validate -pseudoManagementGroup param.

File: patterns/alz/scripts/Start-AMBACleanup.ps1 [hunk 6]
```diff
@@ -11,36 +23,29 @@
...
 .EXAMPLE
-    ./Start-AMBACleanup.ps1 -pseudoManagementGroup Contoso -ReportOnly
-    # generate a list of the resource IDs which would be deleted by this script
+    ./Start-AMBACleanup.ps1 -pseudoManagementGroup Contoso -WhatIf
+    # show output of what would happen if deletes executed.
 
 .EXAMPLE
-    ./Start-AMBACleanup.ps1 -pseudoManagementGroup Contoso -WhatIf
-    # show output of what would happen if deletes executed
+    ./Start-AMBACleanup.ps1 -pseudoManagementGroup Contoso
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Start-AMBACleanup.ps1:35]: Validate param

File: patterns/alz/scripts/Start-AMBAOldArpCleanup.ps1 [hunk 7]
```diff
@@ -12,36 +24,29 @@
...
 
     This script leverages the Azure Resource Graph to find object to delete. Note that the Resource Graph lags behind ARM by a couple minutes.
+
 .LINK
     https://github.com/Azure/azure-monitor-baseline-alerts
 
 .EXAMPLE
-    ./Start-AMBAOldArpCleanup.ps1 -pseudoManagementGroup Contoso -ReportOnly
-    # generate a list of the resource IDs which would be deleted by this script
+    ./Start-AMBAOldArpCleanup.ps1 -pseudoManagementGroup Contoso -WhatIf
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Start-AMBAOldArpCleanup.ps1:32]: Validate param

File: patterns/alz/scripts/Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1 [hunk 8]
```diff
@@ -11,36 +22,29 @@
...
 
     This script leverages the Azure Resource Graph to find object to delete. Note that the Resource Graph lags behind ARM by a couple minutes.
+
 .LINK
     https://github.com/Azure/azure-monitor-baseline-alerts
 
 .EXAMPLE
-    ./Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1 -pseudoManagementGroup Contoso -ReportOnly
-    # generate a list of the resource IDs which would be deleted by this script
+    ./Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1 -pseudoManagementGroup Contoso -WhatIf
```
Comment (arjenhuitema) [inline patterns/alz/scripts/Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1:30]: Validate param

**Review by arjenhuitema — CHANGES_REQUESTED** [review]
Looks good in general just some instances where a param needs to be validated.

File: docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2023-11-14.md (hunk 1 above)
Comment (Brunoga-MS) [reply docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2023-11-14.md]: Done

File: docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2024-03-01.md (hunk 2 above)
Comment (Brunoga-MS) [reply docs/content/patterns/alz/UpdateToNewReleases/Update_from_release_2024-03-01.md]: Done

File: docs/content/patterns/alz/Cleaning-up-a-Deployment.md (hunk 3 above)
Comment (Brunoga-MS) [reply docs/content/patterns/alz/Cleaning-up-a-Deployment.md]: Done

File: patterns/alz/scripts/Remove-AMBADeployments.ps1 (hunk 4 above)
Comment (Brunoga-MS) [reply patterns/alz/scripts/Remove-AMBADeployments.ps1]: Done

File: patterns/alz/scripts/Start-AMBACleanup.ps1 (hunk 6 above)
Comment (Brunoga-MS) [reply patterns/alz/scripts/Start-AMBACleanup.ps1]: Done

File: patterns/alz/scripts/Start-AMBAOldArpCleanup.ps1 (hunk 7 above)
Comment (Brunoga-MS) [reply patterns/alz/scripts/Start-AMBAOldArpCleanup.ps1]: Done

File: patterns/alz/scripts/Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1 (hunk 8 above)
Comment (Brunoga-MS) [reply patterns/alz/scripts/Start-AMBAPolicyInitiativesAndAssignmentsCleanup.ps1]: Done

**Comment by Brunoga-MS** [comment]
Addressed requested changes

**Review by arjenhuitema — APPROVED** [review]

**Pull request merged**